src/components/picker/index.tsx

```diff
@@ -218,21 +235,22 @@ const Picker = React.forwardRef((props: PickerProps, ref) => {
         onCancel: cancelSelect
       };
 
-      return renderCustomModal(modalProps);
+      return renderCustomModal ? renderCustomModal(modalProps) : typeRenderCustomModal?.(modalProps);
     }
   };
 
   const expandableModalContent = useMemo(() => {
-    const useItems = useWheelPicker || propItems;
+    const useItems = isWheelPicker || propItems;
+    const listTopBarProps = type.modal ? headerProps : topBarProps;
     return (
       <PickerItemsList
         testID={`${testID}.modal`}
-        useWheelPicker={useWheelPicker}
+        useWheelPicker={isWheelPicker}
```

Comment:
Why PickerItemsList is not moving to use the picker type?

Reply:
Created new `type` prop that will have the same type as the `usePickerType` returned type value, and will use it to determine the picker type.